### PR TITLE
[NETBEANS-5503] Update JUnit from 4.13.1 to 4.13.2

### PR DESCRIPTION
**Pull request by pepness**
Notes:
- Update all junit4 old references to junit-4.13.2
- Add missing public packages to project.xml file

NetBeans Testing:
- Full build done
- Verify successfull execution of libraries and licenses Ant test
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Checked the files are in the Ant Library Manager
- Clean, build and Test an Ant project with JUnit4 tests
- Check working javadoc and sources
NETBEANS-5503

[JUnit4 Web Page](https://junit.org/junit4/)
[Release Notes](https://github.com/junit-team/junit4/blob/HEAD/doc/ReleaseNotes4.13.2.md)

I will squash the commits before merging.

### CODE REVIEW

File: platform/libs.junit4/nbproject/project.xml [hunk 1]
```diff
@@ -26,24 +26,45 @@
...
+                <package>org.junit.experimental</package>
+                <package>org.junit.experimental.categories</package>
+                <package>org.junit.experimental.max</package>
+                <package>org.junit.experimental.results</package>
+                <package>org.junit.experimental.runners</package>
+                <package>org.junit.experimental.theories</package>
+                <package>org.junit.experimental.theories.internal</package>
+                <package>org.junit.experimental.theories.suppliers</package>
+                <package>org.junit.function</package>
+                <package>org.junit.internal</package>
```
Comment (jtulach) [inline platform/libs.junit4/nbproject/project.xml:46]: Why do you think the module should expose `internal` packages?

Comment (pepness) [reply platform/libs.junit4/nbproject/project.xml]: My bad, I did not check the javadoc to see what packages should be exposed. I have rectified the packages that should be exposed as per the [javadoc](https://junit.org/junit4/javadoc/latest/index.html)

File: apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ModuleList.java [hunk 2]
```diff
@@ -857,7 +857,7 @@ public boolean accept(File pathname) {
         static final String CNB = "org.netbeans.libs.junit4";
         JUnitPlaceholderEntry(File root) {
             super(CNB, new File(root, "platform/modules/" + CNB.replace('.', '-') + ".jar"),
-                    new File[] {new File(System.getProperty("user.home"), ".m2/repository/junit/junit/4.8.2/junit-4.8.2.jar")},
+                    new File[] {new File(System.getProperty("user.home"), ".m2/repository/junit/junit/4.8.2/junit-4.13.2.jar")},
```
Comment (sdedic) [inline apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ModuleList.java:860]: Shouldn't the parent directory's name (seems version of the artifact) change, too ?

Comment (pepness) [reply apisupport/apisupport.ant/src/org/netbeans/modules/apisupport/project/universe/ModuleList.java]: Fixed, thanks.

**Review by sdedic — APPROVED** [review]

**Comment by sdedic** [comment]
@pepness mark with 12.4 milestone, if you want to release it :) for sure.

**Comment by pepness** [comment]
Commits squashed.

**Comment by lkishalmi** [comment]
Well, I'd revert the merge commit and instead of merge, rebase it from master then force push your branch. That way it is far more easy to merge this back to master (with github's rebase and merge)

**Comment by pepness** [comment]
@lkishalmi revert the last commit to recover the full history of activities, i.e., commits?

**Comment by codeschnell** [comment]
I'm not able to figure out the group id ,artifact id and version of my rs2xml.jar file.I have added it to my project made using Add Dependencies in apache netbeans 12.5.Can someone please help me figure it out?

**Pull request merged**